### DISCUSSION TITLE
What command add Encoded name and remove encoding setting on x265 if seen on mediainfo?

### DISCUSSION THREAD

**Discussion by softsubs** [staxrip/staxrip#732] [Q&A] [answered]
example > https://imgur.com/a/63hdWv5

like the picture above I want to add the encoder name and remove the encoding setting but it doesn't affect the other settings. what command do i need to enter?

**Comment by papa-pitufo** [comment]
I don't konw, but you can edit MediaInfo with wordpad (by example), therefore you can modify whatever you want.

**Reply by softsubs** [reply]
How does it work? I don't know this, can you explain and help me.

**Comment by Dendraspis** [comment]
You have to clone a x265 repository and edit it's source code to change `Encoded by`.
You can do the same with `Encoding info`, but you can also remove this info by removing the `Info` checkmark here:

![image](https://user-images.githubusercontent.com/50659978/124471314-034e5000-dd9d-11eb-9742-04fbe1d60abc.png)

**Reply by softsubs** [reply]
"You have to clone a x265 repository and edit it's source code to change Encoded by."

Add encoded name I don't really understand the steps, but I really want to get this result, so can you explain in more detail about this, I encoded with basic settings rip x265 with crf21 it is my favorite, I really look forward to the continuation, thanks before.

Remove encoding setting it worked.

**Reply by Dendraspis** [reply]
You have to clone an x265 repository, change the source code (or just delete the related code lines) to eliminate `Encoded By` and then compile the code to create your own build aka exe file.

"More detailed" would mean doing it for you, sorry.
If you can or do code a bit, it should be doable. If you've never done something like that, you maybe should ask somebody to do it for you.

**Reply by softsubs** [reply]
Thanks for the answer, it's a shame I don't understand that and there's no one I can ask for this other than here :) or maybe you can help me "too much hope :)" sorry

As a last question can you please help me to show this setting because the result of this setting is very good for me, and I really want to get it :)
⤵
Encoding settings : cpuid=1111039 / frame-threads=4 / wpp / no-pmode / no-pme / no-psnr / no-ssim / log-level=2 / input-csp=1 / input-res=1920x1080 / interlace=0 / total-frames=0 / level-idc=0 / high-tier=1 / uhd-bd=0 / ref=6 / no-allow-non-conformance / no-repeat-headers / annexb / no-aud / no-hrd / info / hash=0 / no-temporal-layers / no-open-gop / min-keyint=23 / keyint=250 / gop-lookahead=0 / bframes=8 / b-adapt=2 / b-pyramid / bframe-bias=0 / rc-lookahead=80 / lookahead-slices=6 / scenecut=40 / hist-scenecut=0 / radl=0 / no-splice / no-intra-refresh / ctu=64 / min-cu-size=8 / no-rect / no-amp / max-tu-size=32 / tu-inter-depth=1 / tu-intra-depth=1 / limit-tu=0 / rdoq-level=2 / dynamic-rd=0.00 / no-ssim-rd / signhide / no-tskip / nr-intra=0 / nr-inter=0 / no-constrained-intra / strong-intra-smoothing / max-merge=5 / limit-refs=3 / no-limit-modes / me=3 / subme=4 / merange=57 / temporal-mvp / no-frame-dup / hme / Level / merange / L0,L1,L2=16,32,57 / weightp / weightb / no-analyze-src-pics / deblock=1:-1 / no-sao / no-sao-non-deblock / rd=4 / selective-sao=0 / early-skip / rskip / no-fast-intra / no-tskip-fast / no-cu-lossless / b-intra / no-splitrd-skip / rdpenalty=0 / psy-rd=0.65 / psy-rdoq=1.25 / no-rd-refine / no-lossless / cbqpoffs=-1 / crqpoffs=-1 / rc=crf / crf=21.2 / qcomp=0.60 / qpstep=4 / stats-write=0 / stats-read=0 / ipratio=1.40 / pbratio=1.30 / aq-mode=3 / aq-strength=0.70 / cutree / zone-count=0 / no-strict-cbr / qg-size=32 / no-rc-grain / qpmax=69 / qpmin=0 / no-const-vbv / sar=1 / overscan=0 / videoformat=5 / range=0 / colorprim=2 / transfer=2 / colormatrix=1 / chromaloc=0 / display-window=0 / cll=0,0 / min-luma=0 / max-luma=1023 / log2-max-poc-lsb=8 / vui-timing-info / vui-hrd-info / slices=1 / no-opt-qp-pps / no-opt-ref-list-length-pps / no-multi-pass-opt-rps / scenecut-bias=0.05 / hist-threshold=0.03 / no-opt-cu-delta-qp / no-aq-motion / no-hdr10 / no-hdr10-opt / no-dhdr10-opt / no-idr-recovery-sei / analysis-reuse-level=0 / analysis-save-reuse-level=0 / analysis-load-reuse-level=0 / scale-factor=0 / refine-intra=0 / refine-inter=0 / refine-mv=1 / refine-ctu-distortion=0 / no-limit-sao / ctu-info=0 / no-lowpass-dct / refine-analysis-type=0 / copy-pic=1 / max-ausize-factor=1.0 / no-dynamic-refine / no-single-sei / no-hevc-aq / no-svt / no-field / qp-adaptation-range=1.00 / scenecut-aware-qp=0conformance-window-offsets / right=0 / bottom=0 / decoder-max-rate=0 / no-vbv-live-multi-pass
Default                                  : Yes
Forced                                   : No
Color range                           : Limited
Matrix coefficients                 : BT.709

**Comment by papa-pitufo** [comment]
@softsubs 
I try to explain

You can't erase by defect the phrase "encode by".

All information that you see in your MediaInfo is because you pick up the option "Info".

Its very useful pick up "Info".

When you finisch your work, open this file with MediaInfo, then you select "export" and name this new file.
The new file is txt file, therefore you can edit and erase all that you want.

Looking your preset I suggest:
Uhd-bd: enable
No enable weight-p and -b
Dblock: -3:3
Cutree: OFF
chromaloc=2

Nomore.

Best regards

**Reply by softsubs** [reply]
Hi thanks for your help, I will try it.

**Comment by JJKylee** [comment, marked as answer]
### Copying `encoding settings`

**_Encoding depends heavily on the source_**, so it's practically _no use_ mimicking the encoding settings of a specific file for other sources.

But if you insist on doing so, you need to check

1. rate control(`rc`) mode (`crf` or `bitrate`, etc.)
2. [presets](https://x265.readthedocs.io/en/master/presets.html#presets) and [tunes](https://x265.readthedocs.io/en/master/presets.html#tuning)
3. other settings different from _presets_ and _tunes_ based on the [x265 manual](https://x265.readthedocs.io/en/master/index.html).

in the `Encoding settings` metadata.

This is a very tedious and complicated job and I don't think there's anyone willing to do this for you.

### Encoder name global metadata

You can easily set it in `Container Options` » `Tags` in the case of `mkvmerge`.

![StaxRip_2 6 21_mkvmerge_tags_setting_20210709](https://user-images.githubusercontent.com/3157518/125108566-0bf19f80-e097-11eb-9607-384ee2615b3d.png)

But in the case of `MP4Box`, only pre-defined tags can be added. See [this `MP4Box` manual](https://github.com/gpac/gpac/wiki/mp4box-other-opts#tagging-support).

### Removing `encoding settings`

Only `x264` and `x265` record `encoding settings` metadata in the video stream, and @Dendraspis showed how to disable it in `x265`.

For reference, in the case of `x264`, you cannot disable it at the time of encoding unless you customize the `x264` source yourself and build your own `x264` executable.

However, you can _remove_ the metadata in the existing file using `ffmpeg`. Here's a sample command line code:

```
ffmpeg -i INPUT -map 0 -c copy -bitexact -bsf:v filter_units=remove_types=6 -y OUTPUT
```

That said, there's some catch in this approach. See [this thread](https://forum.videohelp.com/threads/401807-FFmpeg-remove-encoding-settings-from-output) for more info.

**Reply by softsubs** [reply]
"Copying encoding settings"
Yes you are right, because the source is not always the same as your suggestion I will follow it.

"Encoder name global metadata"
surprisingly, it turned out to be that easy. thank you thank you very much

"Removing encoding settings"
Ok. because I haven't tried it with x264 . rarely do I encode x264. for ffmpeg I will try,

all your help means a lot to me, again thank you very much
